(feat): fix has-many docs (#192)
chalk-ai/docs: has-many.mdx

### Many-to-many

The recommended way to define a many-to-many relationship is through a joining feature class.
For instance, to define a many-many relationship between `Actors` and `Movies`, you
could write the following feature classes:

```python
from chalk.features import features, DataFrame

@features
class Actor:
  id: str
  appearances: "DataFrame[ActorInMovie]"
  full_name: str

  # this will be used to demonstrate one of the ways the joining feature can be populated
  movie_ids: list[int]

@features
class Movie:
  id: str
  title: str

@features

> class ActorInMovie:

Reviewer: Maybe this is a `MovieRole`?

  id: str
  actor_id: Actor.id
  movie_id: Movie.id
  movie: Movie
```

<TipInfo>
  Here you need to use quotes around `DataFrame[ActorInMovie]` to use a forward reference.
</TipInfo>

> This joining feature class can either by populated by:
> 
> A SQL Resolver

Reviewer 2: ```suggestion
This joining feature class can be populated by a SQL file resolver:
```

```sql
-- resolves: ActorInMovie
-- source: PG

> SELECT a.id||'_'||m.id as id, a.id as actor_id, m.id as movie_id FROM actors a CROSS JOIN movies m;

Reviewer: not every actor was in every movie, so it's a little weird to me to use a cross-join in this example. could we make this example select from a table like "movie_roles" or "characters"?

```

> or, by a `DataFrame` returning python resolver (namespaced to one of the joined feature sets):

Reviewer 2: ```suggestion
Alternatively, it can be resolved by a Python resolver returning a DataFrame referencing one of the joined feature sets:
```

Author: I merged this with Elliot's suggestion just because technically the DataFrame that is returned references the joined feature set (MovieRole) but is namespaced in one of the other feature sets (Actor or Movie).

> or, by a `DataFrame` returning python resolver (namespaced to one of the joined feature sets):

Reviewer: ```suggestion
or, by a `DataFrame`-returning Python resolver (namespaced to one of the joined feature sets):
```

```python
@online
def get_actor_in_movie(a_id: Actor.id, movie_ids: Actor.movie_ids) -> Actor.appearances:
  return DataFrame([
    ActorInMovie(id=f"{a_id}_{m_id}", actor_id=a_id, movie_id=m_id)
    for m_id in movie_ids
  ])

```

The joining feature class lets you:
- query for movie features from the `Actor` namespace, and
- use movie features in downstream `Actor` resolvers.

> For example, if you wanted to get the titles of the movies an actor appears in, you could run the following query:

Author: ```suggestion
For example, if you wanted to get the titles for all the movies that an actor has appeared in, you could run the following query:
```

> For example, if you wanted to get the titles for all the movies that an actor has appeared in, you could run the following query:

Reviewer 2: ```suggestion
For example, to get the titles for all the movies that an actor has appeared in, you can run the following query:
```

```sh

> chalk query --in actor.id=1 --out actor.appearances.movie.title

Reviewer: maybe show a sample result from running this?

### Many-to-many

The recommended way to define a many-to-many relationship is through a joining feature class.
For instance, if you were defining a many-many relationship between `Actors` and `Movies`, you
would define something like:

```python
from chalk.features import features, DataFrame

@features
class Actor:
  id: str
  appearances: "DataFrame[ActorInMovie]"
  full_name: str

  # this will be used to demonstrate one of the ways the joining feature can be populated
  movie_ids: list[int]

@features
class Movie:
  id: str
  title: str

>   cast: list[ids]

Author: ```suggestion
```
